model/task_lifecycle.go

```diff
@@ -1729,3 +1729,44 @@ func checkResetDisplayTask(t *task.Task) error {
 	}
 	return errors.Wrap(TryResetTask(t.Id, evergreen.User, evergreen.User, details), "resetting display task")
 }
+
+// MarkUnallocatableContainerTasksSystemFailed marks any container task within
+// the candidate task IDs that needs to re-allocate a container but has used up
+// all of its container allocation attempts as finished due to system failure.
+func MarkUnallocatableContainerTasksSystemFailed(candidateTaskIDs []string) error {
+	var unallocatableTasks []task.Task
+	for _, taskID := range candidateTaskIDs {
+		tsk, err := task.FindOneId(taskID)
+		if err != nil {
+			return errors.Wrapf(err, "finding task '%s'", taskID)
+		}
+		if tsk == nil {
+			continue
```

Comment:
Should this also be an error? I'd suspect if we got here from the pod allocator invocaction that this would be unexpected.